parachain/pallets/control/src/lib.rs

```diff
@@ -167,5 +259,29 @@ pub mod pallet {
 			T::OutboundQueue::submit(ticket).map_err(|_| Error::<T>::SubmissionFailed)?;
 			Ok(())
 		}
+
+		fn convert_location(
+			mut location: MultiLocation,
+		) -> Result<(H256, ParaId, MultiLocation), DispatchError> {
```

Comment:
@alistair-singh  We explicit enforce all extrinsics including `CreateAgent` from  `/Parachain(X)` as Vincent mentioned in https://github.com/Snowfork/snowbridge/pull/917#discussion_r1298317752

Reply:
Ah, just to clarify, we must only enforce /Parachain(X) for create_channel and update_channel.

For create_agent, it can accept any xcm origin. This is necessary to support Transact. For example an ink! pallet on an origin parachain must create an agent before it can send transact messages over the bridge.

Reply:
My bad. revert in https://github.com/Snowfork/snowbridge/pull/917/commits/9d279e97a204451f268581f513ebd53f6b5d4bc4